src/momento/responses/vector_index/data/item.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+
+@dataclass
+class Item:
+    """Represents an entry in the vector index."""
+
+    id: str
+    """The id of the item."""
+
+    vector: list[float]
+    """The vector of the item."""
+
+    metadata: dict[str, str | int | float | bool | list[str]] = field(default_factory=dict)
+    """The metadata of the item."""
+
+
+@dataclass
+class ItemWithoutVector:
+    """Represents an entry in the vector index without a vector."""
+
+    id: str
+    """The id of the item."""
+
+    metadata: dict[str, str | int | float | bool | list[str]] = field(default_factory=dict)
+    """The metadata of the item."""
```

Comment:
In a follow-up PR I will refactor `SearchHit`, `SearchAndFetchVectorsHit`, `UpsertItemBatch` to use these as well.